worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,373 @@
+import os
+import typing
+import threading
+
+from typing import List, Set, TextIO, Dict
+from BaseClasses import Item, MultiWorld, Location, Tutorial, ItemClassification
+from worlds.AutoWorld import World, WebWorld
+import settings
+from .Items import get_item_names_per_category, item_table, filler_items, trap_items
+from .Locations import get_locations
+from .Regions import init_areas
+from .Options import YoshisIslandOptions
+from .setup_game import setup_gamevars
+from .Client import YoshisIslandSNIClient
+from .Rules import set_easy_rules, set_normal_rules, set_hard_rules
+from .Rom import LocalRom, patch_rom, get_base_rom_path, YoshisIslandDeltaPatch, USHASH
+
+class YoshisIslandSettings(settings.Group):
+    class RomFile(settings.SNESRomPath):
+        """File name of the Yoshi's Island 1.0 US rom"""
+        description = "Yoshi's Island ROM File"
+        copy_to = "Super Mario World 2 - Yoshi's Island (U).sfc"
+        md5s = [USHASH]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+class YoshisIslandWeb(WebWorld):
+    theme = "ocean"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Yoshi's Island randomizer"
+        "and connecting to an Archipelago server.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Pink Switch"]
+    )
+
+    tutorials = [setup_en]
+
+class YoshisIslandWorld(World):
+    """Yoshi's Island is a 2D platforming game.
+        During a delivery, Bowser's evil ward, Kamek, attacked the stork, kidnapping Luigi and dropping Mario onto Yoshi's Island.
+        As Yoshi, you must run, jump, and throw eggs to escort the baby Mario across the island to defeat Bowser and reunite the two brothers with their parents."""
+    game = "Yoshi's Island"
+    option_definitions = YoshisIslandOptions
+    required_client_version = (0, 4, 4)
+
+    item_name_to_id = {item: item_table[item].code for item in item_table}
+    location_name_to_id = {location.name: location.code for
+                           location in get_locations(None)}
+    item_name_groups = get_item_names_per_category()
+
+    web = YoshisIslandWeb()
+    settings: typing.ClassVar[YoshisIslandSettings]
+    #topology_present = True
+
+    options_dataclass = YoshisIslandOptions
+    options: YoshisIslandOptions
+
+    locked_locations: List[str]
+    location_cache: List[Location]
+    set_req_bosses: str
+    lives_high: int
+    lives_low: int
+    castle_bosses: int
```

Comment:
Some of these are unused. Not sure if they should be removed.

Reply:
i think lives high got phased out a long while ago when I switched to struct packing the lives? thonk